packages/core/src/configuration/Configuration.ts

```diff
@@ -0,0 +1,39 @@
+import { CredentialProvider, Credentials } from './CredentialsProvider';
+import { container, DependencyContainer, injectable } from 'tsyringe';
+
+export interface SdkConfiguration {
+  credentials?: Credentials;
+  bus?: string;
+  api?: string;
+}
+
+export interface ConfigurationOptions extends SdkConfiguration {
+  credentialProviders?: Array<CredentialProvider>;
+}
+
+@injectable()
+export class Configuration {
+  private options: ConfigurationOptions;
+  get container(): DependencyContainer {
+    return container;
+  }
+
+  constructor(options: ConfigurationOptions) {
+    this.options = options;
+    container.register(Configuration, { useValue: this });
+  }
+
+  public get<T extends keyof SdkConfiguration>(key: T): SdkConfiguration[T] {
+    return this.options[key];
+  }
```

Comment:
Avoid using memento/state pattern here. Wrapping the options makes code less clear.